src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -217,7 +217,11 @@ CHIP_ERROR DoEnableNetwork(NetworkInfo * network)
     {
     case NetworkType::kThread:
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+// TODO: On linux, we are using Reset() instead of Detach() to disable thread network, which is not expected.
+// Upstream issue: https://github.com/openthread/ot-br-posix/issues/755
+#if !CHIP_DEVICE_LAYER_TARGET_LINUX
         ReturnErrorOnFailure(DeviceLayer::ThreadStackMgr().SetThreadEnabled(false));
```

Comment:
If the behaviour is platform-dependent, maybe we could call `SetThreadEnabled()` as part of `SetThreadProvision()`?

Reply:
As described in the comments, this special check will be removed after upstream change of ot-br-posix.
However, I think we can introduce a helper function maybe named `ThreadStackMgr::AttachThreadNetwork`?

@bukepo Do you have any idea about this?

Anyway, this is not the topic of this PR -- it just add a test and make minimal changes to existing code to make it available.

Reply:
Can we change the behavior of Reset in ot-br-posix to just do network reset?